jbpm-wb-case-mgmt/jbpm-wb-case-mgmt-backend/src/main/java/org/jbpm/workbench/cm/backend/server/RemoteCaseManagementServiceImpl.java

```diff
@@ -159,5 +168,101 @@ public void removeComment(final String serverTemplateId, final String containerI
+        List<CaseActionSummary> allActionsSummaries = getAllUserTaskActionSummaries(container, caseId, userId);
+
+        List<CaseActionSummary> availableActions = getCaseActionsByStatus(allActionsSummaries, CaseActionStatus.AVAILABLE);
+        availableActions.addAll(getAdHocActions(serverTemplateId, container, caseId));
+        Actions actions = new Actions();
+        actions.setAvailableActions(availableActions);
+        actions.setInProgressAction(getCaseActionsByStatus(allActionsSummaries, CaseActionStatus.IN_PROGRESS));
+        actions.setCompleteActions(getCaseActionsByStatus(allActionsSummaries, CaseActionStatus.COMPLETED_FAIL, CaseActionStatus.COMPLETED_SUCCESS));
+        return actions;
+    }
+
+    public List<CaseActionSummary> getAllUserTaskActionSummaries(String containerId, String caseId, String userId) {
+        List<CaseActionSummary> allActionsSummaries = new ArrayList<>();
+
+        List<String> allStatus = new ArrayList<>();
+        allStatus.addAll(CaseActionStatus.AVAILABLE.getTaskStatuses());
+        allStatus.addAll(CaseActionStatus.IN_PROGRESS.getTaskStatuses());
+        allStatus.addAll(CaseActionStatus.COMPLETED_FAIL.getTaskStatuses());
+        allStatus.addAll(CaseActionStatus.COMPLETED_SUCCESS.getTaskStatuses());
+
+        List<NodeInstance> activeNodes = client.getActiveNodes(containerId, caseId, 0, PAGE_SIZE_UNLIMITED);
+        List<CaseActionSummary> activeNoUserTaskActionsSummaries =
+                activeNodes.stream()
+                        .filter(c -> !Arrays.asList(CaseActionNodeInstanceMapper.NODE_TYPE_HUMAN_TASK_NODE,
+                                CaseActionNodeInstanceMapper.NODE_TYPE_Dynamic_HUMAN_TASK)
+                                .contains(c.getNodeType()))
+                        .map(new CaseActionNodeInstanceMapper())
+                        .map(s -> {
+                            s.setStatus(CaseActionStatus.IN_PROGRESS.getTaskStatuses().get(0));
+                            return s;
+                        }).collect(toList());
+
+        if (!activeNoUserTaskActionsSummaries.isEmpty()) {
+            allActionsSummaries.addAll(activeNoUserTaskActionsSummaries);
+        }
+
+        List<TaskSummary> actionHumanTaskSummaries = client.findCaseTasksAssignedAsBusinessAdministrator(caseId, userId, allStatus, 0, PAGE_SIZE_UNLIMITED, "createdOn", false);
+        if (!actionHumanTaskSummaries.isEmpty()) {
+            allActionsSummaries.addAll(actionHumanTaskSummaries.stream().map(new CaseActionMapper()).collect(toList()));
+        }
+
+        return allActionsSummaries;
+    }
+
+    public List<CaseActionSummary> getAdHocActions(String serverTemplateId, String containerId, String caseId) {
+        List<CaseActionSummary> adHocActions = client.getAdHocFragments(containerId, caseId)
+                .stream()
+                .map(new CaseActionAdHocMapper())
+                .sorted(comparing(CaseActionSummary::getName)).collect(toList());
```

Comment:
I'm not sure if make sense  sorting all the ad-hoc actions by the name, because the result can be show available actions mixed (the stages ad-hoc and the general ad-hoc). For the moment, I remove the sorting and, the available list will show the actions in this sequence: the 'available' human task, the case general ad-hoc (here I removed the sorting) and The case stages ad-hoc. Every subgroup will be shown as they come, without any other sorting